index.rst

```diff
@@ -1,25 +1,493 @@
++---------+-------+---------------------+------+------+-----+-----+
+
+Write Speed
+-----------
+
+Write speed was measured by writing 10,000 messages (with one exception, noted below) as quickly as ts_salobj could write them.
+We also report read speed, but please note that it is not a measure of maximum read speed.
+As long as the reader can keep up with the writer, read speed should be approximately the same as write speed.
+
++--------+------+-------------------+--------------+--------------+----------+
+| System | Acks | Topic             | Write Speed  | Read Speed   | Lost     |
+|        |      |                   |              |              | Messages |
++--------+------+-------------------+--------------+--------------+----------+
+|        |      |                   | messages/s   | messages/s   |          |
++========+======+===================+==============+==============+==========+
+| DDS    | n/a  | summaryState      | 16,553       | 16,580       |      0   |
++--------+------+-------------------+--------------+--------------+----------+
+| DDS    | n/a  | trackTarget       | 13,557       | 13,577       |      0   |
++--------+------+-------------------+--------------+--------------+----------+
+| DDS    | n/a  | forceActuatorData |  1,912       |  1,492       |  2,632   |
++--------+------+-------------------+--------------+--------------+----------+
+| Kafka  | 0    | summaryState      |  4,730       |  4,739       |      0   |
++--------+------+-------------------+--------------+--------------+----------+
+| Kafka  | 0    | trackTarget       |  4,379       |  4,385       |      0   |
++--------+------+-------------------+--------------+--------------+----------+
+| Kafka  | 0    | forceActuatorData |  3,267       |  3,262       |      0   |
++--------+------+-------------------+--------------+--------------+----------+
+| Kafka  | 1    | summaryState      |  2,065       |  2,066       |      0   |
++--------+------+-------------------+--------------+--------------+----------+
+| Kafka  | 1    | trackTarget       |  1,841       |  1,842       |      0   |
++--------+------+-------------------+--------------+--------------+----------+
+| Kafka  | 1    | forceActuatorData |  1,652       |  1,652       |      0   |
++--------+------+-------------------+--------------+--------------+----------+
+
+The fact that some DDS messages were lost simply shows that DDS salobj can write large messages faster than it can read them.
+This is not a cause for concern, as long as we keep up with the actual rate at which messages are written.
+In practice we do not presently lose messages due to slow reading.
+
+In order for the DDS forceActuatorData read test to finish, despite losing data, we wrote 20,000 messages for that test.
+But any number large enough that the reader can read 10,000 messages should give the same results.
+The reader quits as soon as it has seen the specified number of messages, while the writer can keep going.
+
```

Comment:
That is a good point. The latency tests measured 2000 messages at 20 Hz: thus 100 seconds.

The write/read speed tests are much quicker.

In order to run long term I will need to learn how to configure the Kafka broker retention policy to throw out old data quicker. Not a big deal; Kafka is certainly set up to do it.

I think it would be much more interesting to run these tests with:
* The reader/writer running on one machine
* The Kafka components running on another machine
* Preferably while the network is busy doing other things, such as running integration tests.

However, my understanding is that the hardware on the Tucson test stand is fairly limited. So I have no idea how realistic that is. I suppose we could try it on the summit.

Reply:
You could ask Michael, but I thought the idea behind the TTS is to mimic the summit as much as possible (although it is obviously far from perfect) - which makes it the ideal place to put these tests.

Reply:
It might worth noting that we are already running such experiments on both TTS and summit, after all we use Kafka to feed data to the EFD.. the deployed systems use the same JVM etc..

Reply:
I suspect Confluent's own measurements of latency may be useful here (and they agree with ours). I added a link. Thanks to Tiago for finding that!